packages/ilp-packet/src/index.ts

```diff
@@ -229,13 +318,22 @@ export const serializeIlpReject = (json: IlpReject) => {
 export const deserializeIlpReject = (binary: Buffer): IlpReject => {
   const { type, contents } = deserializeEnvelope(binary)
 
-  if (+type !== Type.TYPE_ILP_REJECT) {
+  if (type !== IlpPacketType.Reject) {
     throw new Error('Packet has incorrect type')
   }
 
   const reader = Reader.from(contents)
+
   const code = reader.read(ILP_ERROR_CODE_LENGTH).toString('ascii')
+  if (!isIlpRejectCode(code)) {
+    throw new Error('Invalid code')
+  }
+
   const triggeredBy = reader.readVarOctetString().toString('ascii')
+  if (!isValidIlpAddress(triggeredBy) && triggeredBy !== '') {
```

Comment:
It looks like `interledger/stream-receiver` checks the address on init: https://github.com/interledgerjs/interledgerjs/blob/60b38bd60079d38f6cc3a512eb91e6aa4c90199c/packages/stream-receiver/src/index.ts#L158-L160 and again any time `generateCredentials` is called: https://github.com/interledgerjs/interledgerjs/blob/60b38bd60079d38f6cc3a512eb91e6aa4c90199c/packages/stream-receiver/src/index.ts#L216-L220

I prefer this over validating in `serializeIlpPrepare` since it won't check the same ILP address more than once. `interledger/pay` doesn't validate the outgoing address -- but it could.

Reply:
> Should isValidIlpAddress also be checked in serializeIlpPrepare?

My goal would be to enforce the fields of the Prepare are correct before the Prepare is serialized, so serializing doesn't throw any errors. But since the `IlpPrepare` interface couldn't really be stricter without a breaking change, I was thinking about adding a builder class that would only set an `IlpAddress` as the destination -- so the consumer would have to ensure their address was valid beforehand

> interledger/pay doesn't validate the outgoing address

`pay` also validates the ILP address 😉 

https://github.com/interledgerjs/interledgerjs/blob/8d916c1f95b57f10a37d8d37dca794fd152d583f/packages/pay/src/open-payments.ts#L43-L66

https://github.com/interledgerjs/interledgerjs/blob/8d916c1f95b57f10a37d8d37dca794fd152d583f/packages/pay/src/open-payments.ts#L245-L250

Reply:
> pay also validates the ILP address 

:+1: I missed that, thanks.